api/v0alpha0/accesspolicy_types.go

```diff
@@ -0,0 +1,168 @@
+type AccessPolicySpec struct {
+	// TargetRefs specifies the targets of the AccessPolicy.
+	// An AccessPolicy must target at least one resource.
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	TargetRefs []gwapiv1.LocalPolicyTargetReference `json:"targetRefs"`
+	// Rules defines a list of rules to be applied to the target.
+	// An AccessPolicy must have at least one rule.
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	Rules []AccessRule `json:"rules"`
+}
+
+// AccessRule specifies an authorization rule for the targeted backend.
+// If the tool list is empty, the rule denies access to all tools from Source.
+type AccessRule struct {
+	// Source specifies the source of the request.
+	// +required
+	Source Source `json:"source"`
+	// Tools specifies a list of tools.
+	// +optional
+	Tools []string `json:"tools,omitempty"`
+}
+
+// Source specifies the source of a request.
+//
+// Type must be set to indicate the type of source type.
+// Similarly, either SPIFFE or Serviceaccount can be set based on the type.
+type Source struct {
+
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum=ServiceAccount;SPIFFE
+	// +kubebuilder:validation:Required
+	Type AuthorizationSourceType `json:"type"`
+
+	// spiffe specifies an identity that is matched by this rule.
+	//
+	// spiffe identities must be specified as SPIFFE-formatted URIs following the pattern:
+	//   spiffe://<trust_domain>/<workload-identifier>
+	//
+	// The exact workload identifier structure is implementation-specific.
+	//
+	// spiffe identities for authorization can be derived in various ways by the underlying
+	// implementation. Common methods include:
+	// - From peer mTLS certificates: The identity is extracted from the client's
+	//   mTLS certificate presented during connection establishment.
+	// - From IP-to-identity mappings: The implementation might maintain a dynamic
+	//   mapping between source IP addresses (pod IPs) and their associated
+	//   identities (e.g., Service Account, SPIFFE IDs).
+	// - From JWTs or other request-level authentication tokens.
+	//
+	// Note for reviewers: While this GEP primarily focuses on identity-based
+	// authorization where identity is often established at the transport layer,
+	// some implementations might derive identity from authenticated tokens or sources
+	// within the request itself.
+	//
+	// +optional
+	SPIFFE *AuthorizationSourceSPIFFE `json:"spiffe,omitempty"`
+
+	// ServiceAccount specifies a Kubernetes Service Account that is
+	// matched by this rule. A request originating from a pod associated with
+	// this serviceaccount will match the rule.
+	//
+	// The ServiceAccount listed here is expected to exist within the same
+	// trust domain as the targeted workload. Cross-trust-domain access
+	// should instead be expressed using the `SPIFFE` field.
+	// +optional
+	ServiceAccount *AuthorizationSourceServiceAccount `json:"serviceAccount,omitempty"`
```

Comment:
this is implementation detail, assume it is matched after a k8s token auth. 


Should you document where is the sa extracted? From a jwt or from tls certificate verify

Reply:
+1, I'd further mention it depends on the implementation, not the data plane specifically. Could be jwt, could be mtls certificate, could be a special protocol or IP to SA mapping. Preferably we are not opinionated about how.